File: backend/src/main/java/eu/gillespie/bscthesis/request/Definition.java
package eu.gillespie.bscthesis.request;

import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

import java.util.LinkedList;
import java.util.List;

@ToString
@Getter
@Builder
public class Definition {
    List<String> inputVariable = null;
    // inputConstructor
    List<ConditionalDefinition> conditional = new LinkedList<>();
    StatementTreeVertex otherwise;
}

File: backend/src/main/kotlin/eu/gillespie/bscthesis/request/Definition.kt
package eu.gillespie.bscthesis.request

import eu.gillespie.bscthesis.request.InputConstructor
import eu.gillespie.bscthesis.request.ConditionalDefinition
import java.util.LinkedList
import eu.gillespie.bscthesis.request.StatementTreeVertex

data class Definition (
    var inputVariable: List<String>? = null,
    var inputConstructor: InputConstructor? = null,
    var conditional: List<ConditionalDefinition> = LinkedList(),
    var otherwise: StatementTreeVertex,
)